Backend: REST API docs

dpordomingo opened the pull request:
Depends on https://github.com/src-d/code-annotation/pull/37 and https://github.com/src-d/code-annotation/pull/43

Documents the backend public things &rarr; only the last commits: 468f689, 6431393, 468f689

carlosms commented on server/handler/assignments.go:
```diff
@@ -0,0 +1,27 @@
+
+// GetAssignmentsForUserExperiment returns the experiments for the logged user and a given experiment
+// if these assignments does not already exist, they are created in advance
+func GetAssignmentsForUserExperiment() RequestProcessFunc {
```
Instead of saying it returns something, should it be "returns a function that..."?
Also, do you mean returns the **assignments**? And "these assignments **do** not"

carlosms commented on server/handler/render.go:
```diff
@@ -2,49 +2,64 @@ package handler
+	w.Write(content)
 }
+
+// RequestProcessFunc is the function that takes a http.Request, and returns a serializer.Response and an error
```
**an** http

carlosms commented on server/repository/repositories.go:
```diff
@@ -0,0 +1,77 @@
+
+// ErrNoAssignmentsInitialized is the error returned when are requested the Assignments of a User
+// for a given Experiment, but they has not been yet created
+var ErrNoAssignmentsInitialized = fmt.Errorf("No assignments initialized")
```
is the error returned when **the Assignments of a User** are requested
for a given Experiment, but they **have** not been yet created

carlosms commented on server/repository/users.go:
```diff
@@ -20,11 +22,13 @@ func (r *Users) Create(m *model.User) error {
 
-func (r *Users) Get(id int) (*model.User, error) {
+// Get returns the User identified by the passed ID.
+// If the user does not exists, if returns no user nor error
```
does not **exist**,

dpordomingo commented:
I rebased over other PR dependencies, and added only the following commits:
- 6431393 Explain better how handler.RequestProcessFunc and handler.Get behaves
- 468f689 Fix grammar issues as spotted by @carlosms

Thanks for your review!

smacker commented:
it was my suggestion to make a separate PR to avoid blocking #37 in case there will be many comments.

smacker commented:
@dpordomingo I remember I left a comment about the name of wrapper `Get`. Don't see it now most probably because of rebase. But also don't see any answer for it.

dpordomingo commented:
I also tried to look for it too because I also remembered it, but I could not find so I just disregarded it :face_with_head_bandage:

Thanks for reviewing it again. I'd suggest continuing with it in a separated issue https://github.com/src-d/code-annotation/issues/46
